integration_tests/base_routes.py

```diff
@@ -1103,7 +1109,15 @@ def authenticate(self, request: Request) -> Optional[Identity]:
                 return Identity(claims={"key": "value"})
             return None
 
+    app.include_router(sub_router)
+    app.include_router(di_subrouter)
+    app.include_router(auth_subrouter_endpoint)
+    app.include_router(auth_subrouter_include, auth_required=True)
+    app.include_router(auth_subrouter_instance)
+    app.include_router(auth_subrouter_include_false, auth_required=False)
+    app.include_router(auth_subrouter_include_true, auth_required=True)
```

Comment:
@IA-PieroCV , I believe `auth_required` should either be removed from `include_router` or the `SubRouter()` class